lib/inkdrop-note-templates.js

```diff
@@ -81,6 +92,13 @@ module.exports = {
                 createdAt: +new Date(),
                 updatedAt: +new Date(),
             }
+            // remove Template tag from new note if any.
+            for (let tagIndex = note.tags.length - 1; tagIndex >= 0; tagIndex--) {
```

Comment:
Probably, we can write this filtering logic like the following.

```js
const templateTag = await db.tags.findWithName(TEMPLATE_TAG_NAME);
const tagsWithoutTemplate = templateTag ? template.tags.filter(tag => tag !== templateTag._id) : [];
const note = {
    ...template.doc,
    _id: db.notes.createId(),
    bookId: currentBookId,
    _rev: undefined,
    title: template.title,
    body: template.body,
    tags: tagsWithoutTemplate,
    createdAt: +new Date(),
    updatedAt: +new Date(),
}
```

https://docs.inkdrop.app/reference/db-tag#findwithnamename

I want to avoid [N+1 problem](https://stackoverflow.com/questions/97197/what-is-the-n1-selects-problem-in-orm-object-relational-mapping).

Reply:
I totally agree with you on this aspect. Let me modify the implementation.

Reply:
Consider the following case.
 
1. User has no `Template` tag.
2. The template has two tags: `["2021", "Life"]`.

By executing your suggested code, `templateTag` is `undefined` and `tagsWithoutTemplate` is `[]`. However, I think users may want to preserve the tags instead of clearing all tags. Thus, I make last part of ternary operator as `template.tags` itself without filtering like below code.

```javascript
const tagsWithoutTemplate = templateTag ? template.tags.filter(tag => tag !== templateTag._id) : template.tags;
```

How do you think about this?

Reply:
There's a relating bug fixing in commit a7c7839. Please check it.

Reply:
>  I make last part of ternary operator as `template.tags` itself without filtering like below code.

Nice!